public/app/percona/integrated-alerting/components/Table/Filter/Filter.tsx

```diff
@@ -142,15 +142,11 @@ export const Filter = ({ columns, rawData, setFilteredData, hasBackendFiltering
           </div>
           {showAdvanceFilter && openCollapse && (
             <div className={styles.advanceFilter}>
-              {columns.map((column) => {
-                if (column.type === FilterFieldTypes.DROPDOWN) {
-                  return <SelectDropdownField column={column} />;
-                }
-                if (column.type === FilterFieldTypes.RADIO_BUTTON) {
-                  return <RadioButtonField column={column} />;
-                }
-                return <></>;
-              })}
+              {columns.map(
+                (column) =>
+                  (column.type === FilterFieldTypes.DROPDOWN && <SelectDropdownField column={column} />) ||
+                  (column.type === FilterFieldTypes.RADIO_BUTTON && <RadioButtonField column={column} />)
```

Comment:
Is this `||` necessary? I think it's doable without it

Reply:
```
(column) => 
    <>
        {column.type === FilterFieldTypes.DROPDOWN && <SelectDropdownField column={column} />}
        {column.type === FilterFieldTypes. RADIO_BUTTON && <RadioButtonField column={column} />}
    </>
```

Reply:
But this will add this logic to every column and then after the map is finished react needs to decide on every column what he will show. I am not sure if this is a good approach and probably in imaginary cases when we have many columns this can be bad for performance, can't it?

Reply:
Good point, maybe it's not a great idea within a map 👍